src/main/java/io/confluent/connect/elasticsearch/ElasticsearchWriter.java

```diff
@@ -211,6 +211,22 @@ public void write(Collection<SinkRecord> records) {
 
       bulkProcessor.add(indexableRecord, flushTimeoutMs);
     }
+
+  }
+
+  /**
+   * Return the expected index name for a given topic. It makes sure to lowercase the topic name
+   * as elasticsearch require all index names to be lowercase.
+   * See https://github.com/confluentinc/kafka-connect-elasticsearch/issues/246 for connector details.
+   * See https://github.com/elastic/elasticsearch/issues/29420 for elasticsearch details.
+   * @param topic The topic name being proceed.
+   * @return String A valid elasticsearch index name
+   */
+  private String convertTopicToIndexName(String topic) {
+    final String indexOverride = topicToIndexMap.get(topic);
+    String index = indexOverride != null ? indexOverride : topic.toLowerCase();
```

Comment:
```suggestion
    String index = indexOverride != null ? indexOverride : topic.toLowerCase(StandardCharsets.UTF_8);
```

Reply:
lowercase accepts locale, but not Charset, right? we might like to add this kind of encoding validation before we reach here, isn't?

Reply:
Oh, sorry. My bad. And this is actually not required, so it's okay to not change this line.